crates/sui-core/src/authority_server.rs

```diff
@@ -501,8 +502,9 @@ impl ValidatorService {
                     &self.state.name,
                     certificate.clone().into(),
                 );
-                self.consensus_adapter.submit(
-                    transaction,
+                let transactions = vec![transaction];
+                self.consensus_adapter.submit_batch(
+                    transactions,
```

Comment:
submit_batch should take `&[ConsensusTransaction]` instead of `&Vec<ConsensusTransaction>`, then there is no need to create a temporary vector here.